### PR TITLE
SAT: all `additionalProperties` fields must be set to `true`

### LINKED ISSUE #14600: SAT: Assert that `additionalProperties: false` is not specified in json schemas
(more context in parent issue https://github.com/airbytehq/airbyte/issues/14097)

Following up on https://github.com/airbytehq/airbyte/issues/14196, to encode this going forward we should add checks in the SAT to prevent `additionalProperties: false`.  

From the [JSON Schema docs](http://json-schema.org/understanding-json-schema/reference/object.html#additional-properties), `additionalProperties` does not necessarily have to be a boolean, so we should just check if it's set and its value is anything other than `true`.

## Acceptance criteria

- A connector spec schema with `additionalProperties` set to anything other than `true` fails SAT
- A stream schema with `additionalProperties` set to anything other than `true` fails SAT

### PR DESCRIPTION
**Pull request by alafanechere**
## What
Closing https://github.com/airbytehq/airbyte/issues/14600
Add tests to make sure all the connector setting the `additionalProperties` field in their spec or stream schema set the value of this field to `true`. Check https://github.com/airbytehq/airbyte/issues/14196 to understand why this is required.

## How 
* Add a [`find_all_values_for_key_in_schema`](https://github.com/airbytehq/airbyte/blob/2581129239ed707b88fc83365a050ed5c65311d3/airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/utils/common.py#L122) function to retrieve all the values for a specific field in a schema. ([And unit test it](https://github.com/airbytehq/airbyte/blob/2581129239ed707b88fc83365a050ed5c65311d3/airbyte-integrations/bases/source-acceptance-test/unit_tests/test_utils.py#L377))
* Add two tests to `test_core.py` using this function to check all the values for the `additionalProperties` field are `true`:
  * [In TestSpec](https://github.com/airbytehq/airbyte/blob/2581129239ed707b88fc83365a050ed5c65311d3/airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/tests/test_core.py#L168)
  * [ In TestDiscovery](https://github.com/airbytehq/airbyte/blob/2581129239ed707b88fc83365a050ed5c65311d3/airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/tests/test_core.py#L266)

## 🚨 User Impact 🚨
I think this will cause a lot of build failures because a lot of connectors are not following this rule and use `"additionalProperties" : false`.

**I think we also need to update our tutorials and code generation template (`*.hbs`) that are using `"additionalProperties": false`**.[ I created an issue for this.](https://github.com/airbytehq/airbyte/issues/14879)

### CODE REVIEW

**Comment by alafanechere** [comment]
I'm going to test if running test with the slash command uses a dev image of SAT and run a test for connectors that whose acceptance tests should failed on:
* source-open-weather's spec has `"additionalProperties": false`
* source-amazon-seller-partner has streams with `"additionalProperties": false`

**Comment by alafanechere** [comment]
☝️ Both test failures above are expected and show that the new tests work as expected.

**Review by pedroslopez — APPROVED** [review]
Nice job 👍🏼 

There's an existing helper method that could have been used here, but I'll leave it up to you if you want to move over to that or keep the new one. My initial thought would be to use one or the other since they're so similar.

With regards to this change breaking builds for many connectors, do you know how many (python) sources are affected? We should check in with the dbs/dws team on priority for this existing issue #14718. If it will be done soon then maybe we hold off as to not impact publishes, if not then we could prioritize GA/Beta connectors and do those first.

File: airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/utils/common.py
Comment (alafanechere) [reply airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/utils/common.py:129]: Yep I tried `find_key_inside_schema`, it's indeed working but only returns the first schema item having the searched key. I could have implemented a recursion in the final test function to use this function. But I preferred to use my new function to defer the recursion to the helper to boost the readability of the test.  And this function can be helpful in the future if we have other keys we want to validate value on.

**Comment by alafanechere** [comment]
> My initial thought would be to use one or the other since they're so similar.

I [replaced find_key_inside_schema with find_all_values_for_key_in_schema](https://github.com/airbytehq/airbyte/pull/14878/commits/edce556f83f155315991a926279dba332e9f9a48) in tests checking for `$ref`. Original unit tests on functions that used `find_key_inside_schema` are still passing.

**Comment by alafanechere** [comment]
> do you know how many (python) sources are affected

I created a list I shared in https://github.com/airbytehq/airbyte/issues/14718

**Comment by alafanechere** [comment]
I made the required changes to beta and GA connectors in https://github.com/airbytehq/airbyte/pull/14924

**Comment by alafanechere** [comment]
☝️ The test above should now pass after https://github.com/airbytehq/airbyte/pull/14924 was merged

**Comment by alafanechere** [comment]
/publish connector=bases/source-acceptance-test auto-bump-version=false

**Comment by alafanechere** [comment]
/publish connector=bases/source-acceptance-test auto-bump-version=false

**Pull request merged**
